base/expr.jl

```diff
@@ -1333,3 +1333,60 @@ function make_atomicreplace(success_order, fail_order, ex, old_new)
         return :(replaceproperty!($ll, $lr, $old_new::Pair..., $success_order, $fail_order))
     end
 end
+
+"""
+    @atomiconce a.b.x = value
+    @atomiconce :sequentially_consistent a.b.x = value
+    @atomiconce :sequentially_consistent :monotonic a.b.x = value
+
+Perform the conditional assignment of value atomically if it was previously
+unset, returning the value `success::Bool`. Where `success` indicates whether
```

Comment:
So one thing is that unset isn't the nomenclature we seem to usually use, we mostly seem to use defined/undefined

Reply:
I guess that should say `the field was not set` or `the field was undef`?

Reply:
I think that might be a bit more inline to what we call it in other places?

Reply:
We use unset at lot for this also, since we use undefined to mean isbits types that are not initialized, while this is not undef->defined but rather unset->set